### ISSUE TITLE
'NoneType' object has no attribute 'add_step' (Trac #139)

### ISSUE THREAD

**Issue by cfobel** [wheeler-microfluidics/microdrop#143]
Error when actuating an electrode in realtime mode for a freshly imported device (i.e., before the device has been saved). This happens because we're trying to add a step to an experiment log that doesn't exist yet.

Error message:

[ERROR]: microdrop.gui.protocol_controller plugin crashed processing on_step_options_changed signal.
Reason: 'NoneType' object has no attribute 'add_step'
[INFO]: Traceback (most recent call last):
  File "C:\Users\ryan\Documents\dev\python\microdrop\microdrop\plugin_manager.py", line 453, in emit_signal
    return_codes[observer.name] = f(*args)
  File "C:\Users\ryan\Documents\dev\python\microdrop\microdrop\gui\protocol_controller.py", line 385, in on_step_options_changed
    self.run_step()
  File "C:\Users\ryan\Documents\dev\python\microdrop\microdrop\gui\protocol_controller.py", line 332, in run_step
    app.experiment_log.add_step(app.protocol.current_step_number,
AttributeError: 'NoneType' object has no attribute 'add_step'

[ERROR]: microdrop.gui.protocol_controller plugin crashed processing on_step_options_changed signal.
Reason: 'NoneType' object has no attribute 'add_step'

Migrated from http://microfluidics.utoronto.ca/microdrop/ticket/139
```json
{
    "status": "closed", 
    "changetime": "2014-04-17T19:39:43", 
    "description": "Error when actuating an electrode in realtime mode for a freshly imported device (i.e., before the device has been saved). This happens because we're trying to add a step to an experiment log that doesn't exist yet.\n\nError message:\n\n[ERROR]: microdrop.gui.protocol_controller plugin crashed processing on_step_options_changed signal.\nReason: 'NoneType' object has no attribute 'add_step'\n[INFO]: Traceback (most recent call last):\n  File \"C:\\Users\\ryan\\Documents\\dev\\python\\microdrop\\microdrop\\plugin_manager.py\", line 453, in emit_signal\n    return_codes[observer.name] = f(*args)\n  File \"C:\\Users\\ryan\\Documents\\dev\\python\\microdrop\\microdrop\\gui\\protocol_controller.py\", line 385, in on_step_options_changed\n    self.run_step()\n  File \"C:\\Users\\ryan\\Documents\\dev\\python\\microdrop\\microdrop\\gui\\protocol_controller.py\", line 332, in run_step\n    app.experiment_log.add_step(app.protocol.current_step_number,\nAttributeError: 'NoneType' object has no attribute 'add_step'\n\n[ERROR]: microdrop.gui.protocol_controller plugin crashed processing on_step_options_changed signal.\nReason: 'NoneType' object has no attribute 'add_step'", 
    "reporter": "ryan", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1397763583719301", 
    "component": "microdrop", 
    "summary": "'NoneType' object has no attribute 'add_step'", 
    "priority": "minor", 
    "keywords": "", 
    "version": "0.1", 
    "time": "2013-04-16T17:28:51", 
    "milestone": "Microdrop 1.0", 
    "owner": "", 
    "type": "defect"
}
```

**Comment by cfobel** [comment]
Trac update at 20130416T17:34:24: 
* ryan@fobel.net commented:

> commit 9dc0715a614459c326e2a67590b5edb351f06940
> Author: Ryan Fobel <ryan@fobel.net>
> Date:   Tue Apr 16 13:16:02 2013 -0400
> 
>     Fixes #143


* ryan@fobel.net changed resolution from "" to "fixed"
* ryan@fobel.net changed status from "new" to "closed"

**Comment by cfobel** [comment]
Trac update at 20140417T19:39:43: ryan changed component from "" to "microdrop"
